### PR TITLE
feat: Loading component for initialization and fetching

### PR DESCRIPTION
**Pull request by Canuckaholic**
## Description

Creation of a "loading" component for handling the initial page load and also when a fetch of data is happening and the UI needs to refresh.

Related to https://github.com/c6o/node-monorepo/issues/760

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by nsainaney — CHANGES_REQUESTED** [review]

File: packages/web/src/loaders/loading.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ import { cssReboot, cssBase, cssLayouts, cssTypography } from '@c6o/ui-theme'
 @customElement('c6o-loading')
```
[the remark this answers is not in the document]
Comment (Canuckaholic) [reply packages/web/src/loaders/loading.ts:5]: we already have `c6o-progress-bar`. How about `c6o-initialization`?

File: packages/web/src/animations/loading.ts [hunk 2]
```diff
@@ -5,6 +5,9 @@ import { cssReboot, cssBase, cssLayouts, cssTypography } from '@c6o/ui-theme'
 @customElement('c6o-loading')
 export class Loading extends MobxLitElement {
 
+    @property({ type: Boolean, attribute: 'bar-only' })
+    barOnly = false
+
```
Comment (nsainaney) [inline packages/web/src/animations/loading.ts:10]:
```suggestion
    @property({ type: Boolean, attribute: 'simple' })
    simpleUI = false

```

File: packages/web/src/store/panel.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import { MobxLitElement } from '@adobe/lit-mobx'
+
+@customElement('c6o-store-panel')
+export class StorePanel extends MobxLitElement {
+
+    @property({ type: Object })
+    store: EntityStore | EntityListStore
+
+    @property({ type: String })
+    text = 'Loading...'
```
Comment (nsainaney) [inline packages/web/src/store/panel.ts:11]:
```suggestion
    loadingMessage = 'Loading...'
```

File: packages/web/src/store/state.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    barOnly = false
+
+    @property({ type: Boolean })
+    container = false
+
+    @property({ type: Object })
+    store: EntityStore | EntityListStore
+
+    @property({ type: String })
+    text = 'Loading...'
```
Comment (nsainaney) [inline packages/web/src/store/state.ts:19]:
```suggestion
    loadingMessage = 'Loading...'
```

File: packages/web/src/store/state.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
+import { html, customElement, property, css, CSSResult } from 'lit-element'
+import { MobxLitElement } from '@adobe/lit-mobx'
+import { EntityStore, EntityListStore } from '@c6o/common'
+import { cssReboot, cssBase, cssGrid } from '@c6o/ui-theme'
+
+@customElement('c6o-store-state')
+export class StoreState extends MobxLitElement {
+
+    @property({ type: Boolean, attribute: 'bar-only' })
+    barOnly = false
```
Comment (nsainaney) [inline packages/web/src/store/state.ts:10]:
```suggestion
    @property({ type: Boolean, attribute: 'simple' })
    simpleUI = false
```

**Review by nsainaney — CHANGES_REQUESTED** [review]

**Review by nsainaney — APPROVED** [review]

**Pull request merged**